File: Medium/Python/Other/LeastInterval.py
class Solution(object):
    def leastInterval(self, tasks, n):
        """
            :type tasks: List[str]
            :type n: int
            :rtype: int
            """
        tc = collections.Counter(tasks).values()
        m = max(tc)
        mc = tc.count(m)
        return max(len(tasks), (m - 1) * (n + 1) + mc)

File: Medium/Java/Others/LeastInterval.java
class Solution {
    public int leastInterval(char[] tasks, int n) {
        int[] count = new int[26];
        for (char task: tasks)
            count[task - 'A'] ++;
        Arrays.sort(count);
        int mc = 1;
        for (int i = 25; i > 0; i --) {
            if (count[i] == count[i - 1])
                mc ++;
            else break;
        }
        return Math.max(tasks.length, (count[25] - 1) * (n + 1) + mc);
    }
}
